README.md
# echOpen processing tweets (ept tool)
## What ?
![Building the Graph](images/EPT-Thumb.png)

## Some info

Using the graph. Get all the tweets from gdrive to tsv files (an archive is provided)

> usage : python ept.py tweets.tsv > graph.gdf

Remove all echopenorg mentions? Just 

> sed "/@echopenorg/d" ./graph.gdf > ./graph_clean.gdf

# Outputs

* **20160221-Graphe-echOpen.gephi** is the gephi file.. you can directly open it with Gephi.
* **20160221-Graphe-echOpen.pdf** is the PDF -- text-searchable
* **20160221-Graphe-echOpen.png** is a raw image for the graph
							

# Screenshots 
## Building the graph with gephi
![Building the Graph](images/EPT-Work.png)
## Measuring metrics of the graph
![Playing with Graph Data](images/EPT-GraphData.png)
## Displaying the graph
![Displaying the Data](images/EPT-Vizu.png)


Patched up together by @kelu124 to help visualize echopen.org twittosphere


history.txt
    1  cd /m
    2  cd /mnt
    3  ls
    4  mkdir cd
    5  cd
    6  ls
    7  sudo su
    8  mkdir /mnt/cd
    9  cd /mnt/cd
   10  ls
   11  cd ..
   12  mount /dev/cdrom cd
   13  sudo mount /dev/cdrom cd
   14  cd cd
   15  ls
   16  cd /home/kelu/
   17  ls
   18  cd Des
   19  cd Bureau/
   20  ls
   21  cd vmware-tools-distrib/
   22  ls
   23  sudo su
   24  ls
   25  wget http://tools.wmflabs.org/pywikibot/core_stable.zip
   26  unzip core_stable.zip
   27  mv core_stable pywikibots
   28  ls
   29  mv core_stable.zip Bureau/
   30  ls
   31  aptitude update
   32  sudo su
   33  chromium-browser &
   34  ls
   35  ssh root@192.168.1.14
   36  ssh root@92.128.39.83
   37  ssh root@192.168.1.14
   38  ls
   39  cd echOpen/
   40  ls
   41  cd rechOpen/
   42  ls
   43  cd Applications/
   44  ls
   45  cd scope+id
   46  cd scope+pid/
   47  ls
   48  make
   49  ls
   50  cd src/
   51  ls
   52  cd ..
   53  ls
   54  ls -alh
   55  make CROSS_COMPILE=arm-linux-gnueabi- clean all
   56  cd ..
   57  ls
   58  cd ..
   59  ls
   60  cd Applications/
   61  ls
   62  cd scope+gen
   63  ls
   64  make CROSS_COMPILE=arm-linux-gnueabi- clean all
   65  ls
   66  ls -alh
   67  cd ..
   68  ls
   69  cd freqanalyzer/
   70  LS
   71  ls
   72  make CROSS_COMPILE=arm-linux-gnueabi- clean all
   73  cd ..
   74  ls
   75  cd ecosystem/
   76  ls
   77  cd ..
   78  cd assets/
   79  ls
   80  cd ..
   81  ls
   82  cd spectrum/
   83  ls
   84  make CROSS_COMPILE=arm-linux-gnueabi- clean all
   85  ls
   86  cd ..
   87  ls
   88  cd freqanalyzer/
   89  ls
   90  make CROSS_COMPILE=arm-linux-gnueabi- clean all
   91  cd ..
   92  ls
   93  cd ..
   94  ls
   95  git add *
   96  git commit -m "First compilation of web apps"
   97  git push
   98  ssh pitaya
   99  ssh root@192.168.1.14
  100  ls
  101  cd Applications/
  102  ls
  103  history | grep scp
  104  ls
  105  scp -r freqanalyzer root@<red_pitaya_ip>:/opt/www/apps/
  106  scp -r freqanalyzer root@192.168.1.14:/opt/www/apps/
  107  ls
  108  scp -r scope+gen root@192.168.1.14:/opt/www/apps/
  109  ls
  110  scp -r spectrum root@192.168.1.14:/opt/www/apps/
  111  sudo su
  112  scp -r scope+pid root@192.168.1.14:/opt/www/apps/
  113  ls
  114  scp -r scope root@192.168.1.14:/opt/www/apps/
  115  ls
  116  cd
  117  ls
  118  cd echOpen/rechOpen/
  119  ls
  120  cd Applications/
  121  ls
  122  cd calgenscope/
  123  ls
  124  cd src
  125  ls
  126  cd ..
  127  ping google.com
  128  ping 192.168.1.1
  129  cd ..
  130  scp -r scope root@192.168.1.14:/opt/www/apps/
  131  scp -r root@192.168.1.14:/opt/www/apps/calgenscope ./
  132  scp -r root@192.168.1.14:/opt/www/apps/scope ./
  133  ssh root@192.168.1.14
  134  scp -r root@192.168.1.14:/opt/www/apps/lcr_meter ./
  135  ls
  136  cd ..
  137  ls
  138  git add *
  139  git commit -m "New apps"
  140  git push
  141  ls
  142  mkdir Puneet2895
  143  cd Puneet2895/
  144  git clone git@github.com:Puneet2895/redpitaya.git
  145  ls
  146  mv redpitaya/* ./
  147  ls
  148  rm -rf .git
  149  ls
  150  rm -rf redpitaya
  151  ls
  152  more FFT.m 
  153  more 100dump_copy.sh 
  154  ls
  155  cd ..
  156  ls
  157  vi README.md
  158  git add *
  159  git commit -m "Coming from Puneet2895/RedPitaya"
  160  git push
  161  ls
  162  cd Applications/
  163  l
  164  ls
  165  ssh root@192.168.1.14
  166  scp -r root@192.168.1.14:/opt/www/apps/calgenscope ./
  167  ls
  168  cd ..
  169  git add *
  170  find rp.c
  171  find ./ rp.c
  172  find ./ | grep rp.c
  173  find ./ | grep rp*.c
  174  find ./ | grep rp.h
  175  cd ./SDK/include/
  176  ls
  177  more rp.h
  178  ls
  179  cd ..
  180  ls
  181  cd
  182  arp -t
  183  arp -a
  184  arp -a wlan0
  185  ls
  186  make CROSS_COMPILE=arm-linux-gnueabi- clean all
  187  cd ..
  188  scp controller* root:192.168.1.12:/opt/www/apps/echopen/
  189  scp controller* root@192.168.1.12:/opt/www/apps/echopen/
  190  ssh root@192.168.1.12
  191  ls
  192  cd echOpen/
  193  ls
  194  cd rechOpen/
  195  ls
  196  cd Applications/
  197  ls
  198  cd scope
  199  ls
  200  cd src
  201  ls
  202  gedit worker.c
  203  ls
  204  cd ..
  205  make CROSS_COMPILE=arm-linux-gnueabi- clean all 
  206  ls
  207  scp root@192.168.1.14:/opt/www/apps/ controller*
  208  scp root@192.168.1.14:/opt/www/apps/ controller
  209  scp root@192.168.1.14:/opt/www/apps/ controller.so
  210  scp root@192.168.1.1412 /opt/www/apps/ controller.so
  211  scp root@192.168.1.1412 /opt/www/apps/ ./controller.so
  212  scp root@192.168.1.12 /opt/www/apps/ ./controller.so
  213  scp controller.so root@192.168.1.12:/opt/www/apps/
  214  scp controllerhf.so root@192.168.1.12:/opt/www/apps/
  215  gedit src/worker.c
  216  make CROSS_COMPILE=arm-linux-gnueabi- clean all 
  217  scp controller.so root@192.168.1.12:/opt/www/apps/
  218  ls
  219  ls -alh
  220  scp controller* root@192.168.1.12:/opt/www/apps/echopen/
  221  gedit src/worker.c
  222  make CROSS_COMPILE=arm-linux-gnueabi- clean all 
  223  ls
  224  ls -alh
  225  scp controller* root@192.168.1.12:/opt/www/apps/echopen/
  226  ssh 192.168.1.12
  227  ssh root@192.168.1.12
  228  ls
  229  cd src/
  230  ls
  231  cd ..
  232  make clean al
  233  make CROSS_COMPILE=arm-linux-gnueabi- clean all 
  234  cd src
  235  cd ..
  236  gedit main.c
  237  gedit src/main.c
  238  make CROSS_COMPILE=arm-linux-gnueabi- clean all 
  239  scp controller* root@192.168.1.12:/opt/www/apps/echopen/
  240  ssh root@192.168.1.12
  241  ls
  242  cd doc/
  243  ls
  244  cd ..
  245  ls
  246  cd src/
  247  ls
  248  cd ..
  249  ls
  250  cd ..
  251  ls
  252  scp -r root@192.168.1.14:/opt/www/apps/echopen ./
  253  scp -r root@192.168.1.14:2opt/www/apps/echopen ./
  254  scp -r root@192.168.1.12:/opt/www/apps/echopen ./
  255  ls
  256  cd ec
  257  cd echopen/
  258  l
  259  ls
  260  cd ../../../
  261  ls
  262  cd rechOpen/
  263  git add *
  264  git -m commit "echopen app added"

ept.py
#echOpen processing tweets
#usage : python ept.py tweets.tsv > graph.gdf

from math import *
import sys
import string
import re


try:
    sys.argv[1]
except NameError:
    eptFile = 'Missing an arg'
else:
    eptFile = sys.argv[1]

paireds = []
tHList = []
tUsers = []
tHashes = []
i = 0

with open(eptFile, 'r') as Tweets:
	for line in Tweets:
		line = line.split('\t')
		del line[-1]
		#print line

		usered = line[1].lower()

		tmp = line[2].lower()
		tmp = tmp.lower()
		users = []
		users = re.findall("@\w+", tmp)
	
		tHashers = re.findall("#\w+", tmp)
		users.append(usered)
	
		users = users + tHashers 
		tHashes.append(tHashers)
		#print users
		paireds.append(users)
		#paireds.append(tHashers)

		i = i +1 
		tUsers.append(usered)
		tHList = tHList + tHashers

#print paireds
#print tUsers
tUserUnique = []
[tUserUnique.append(item) for item in tUsers if item not in tUserUnique]
tH = []
[tH.append(item) for item in tHList if item not in tH]
#print tUserUnique

print 'nodedef> name VARCHAR,label VARCHAR'
for user in tUserUnique:
	print "%s, %s" % (user, user)
for user in tH:
	print "%s, %s" % (user, user)

print 'edgedef> node1 VARCHAR,node2 VARCHAR'
for seque in paireds:
	l = len(seque)
	#print seque
	#print l
	if l>1:
		L = 0
		while L < l-1:
			#print L
			#print seque[L]
			#print seque[b]
			LL = L + 1
			
			while LL < l:
				print "%s, %s" % ( seque[L], seque[LL] )
				LL = LL + 1
			L = L+ 1
	




t-ec.py
#!/usr/bin/python
import sys
from TwitterSearch import *
# Requis pour MySQL connection
# Requis pour MySQL connection
import MySQLdb
from datetime import datetime


def remo_na(text):
    return ''.join(i for i in text if ord(i)<128)

db=MySQLdb.connect() 
cursor=db.cursor()

add_tweet = ("INSERT INTO tweets "
               "(tWho, tWhat, tDate, tUID) "
               "VALUES (%s, %s, %s, %s)")

# USAGE
# python patent.py US3541840 US3714817 US3864660

try:
    tuo = TwitterUserOrder('echopenorg') # create a TwitterUserOrder


    # it's about time to create a TwitterSearch object with our secret tokens
    ts = TwitterSearch(
        consumer_key = '',
        consumer_secret = '',
        access_token = '',
        access_token_secret = ''
     )

    for tweet in ts.search_tweets_iterable(tuo):
        twText = " "
	twText = remo_na(tweet['text'])

	twID = tweet['id']
	#print '%s' % tweet['created_at']
	tWho = '@' + tweet['user']['screen_name']
	created = datetime.strptime(tweet['created_at'], '%a %b %d %H:%M:%S +0000 %Y')

	query = ("SELECT tID FROM tweets "
	"WHERE tUID LIKE %s ")
	numrows = cursor.execute(query, twID)
	if not(numrows): # (tWho, tWhat, tDate, tUID) 
		data_tweet = (tWho.decode('unicode_escape').encode('ascii','ignore'), twText.encode('ascii', 'ignore'), created, twID )
		cursor.execute(add_tweet, data_tweet)
	if numrows:
		print "Tweet deja enregistre"




except TwitterSearchException as e: # take care of all those ugly errors if there are some
    print(e)

		





db.commit()
cursor.close()
db.close()



t-fa.py
#!/usr/bin/python
import sys
from TwitterSearch import *
# Requis pour MySQL connection
# Requis pour MySQL connection
import MySQLdb
from datetime import datetime
import os


from tecpass import * 

    
def remo_na(text):
    return ''.join(i for i in text if ord(i)<128)



cursor=db.cursor()

add_tweet = ("INSERT INTO tweets "
               "(tWho, tWhat, tDate, tUID) "
               "VALUES (%s, %s, %s, %s)")

# USAGE
# python patent.py US3541840 US3714817 US3864660

try:
    tuo = TwitterUserOrder('echopenorg') # create a TwitterUserOrder


    # it's about time to create a TwitterSearch object with our secret tokens


    for tweet in ts.search_tweets_iterable(tuo):
        twText = " "
	twText = remo_na(tweet['text'])

	twID = tweet['id']
	#print '%s' % tweet['created_at']
	tWho = '@' + tweet['user']['screen_name']
	created = datetime.strptime(tweet['created_at'], '%a %b %d %H:%M:%S +0000 %Y')

	query = ("SELECT tID FROM tweets "
	"WHERE tUID LIKE %s ")
	numrows = cursor.execute(query, twID)
	if not(numrows): # (tWho, tWhat, tDate, tUID) 
		data_tweet = (tWho.decode('unicode_escape').encode('ascii','ignore'), twText.encode('ascii', 'ignore'), created, twID )
		cursor.execute(add_tweet, data_tweet)
	if numrows:
		print "Tweet deja enregistre"




except TwitterSearchException as e: # take care of all those ugly errors if there are some
    print(e)

		





db.commit()
cursor.close()
db.close()



t-hu.py
#!/usr/bin/python
import sys
from TwitterSearch import *
# Requis pour MySQL connection
# Requis pour MySQL connection
import MySQLdb
from datetime import datetime


def remo_na(text):
    return ''.join(i for i in text if ord(i)<128)

db=MySQLdb.connect() 
cursor=db.cursor()

add_tweet = ("INSERT INTO tweets "
               "(tWho, tWhat, tDate, tUID) "
               "VALUES (%s, %s, %s, %s)")

# USAGE
# python patent.py US3541840 US3714817 US3864660

try:
    tso = TwitterSearchOrder() # create a TwitterSearchOrder object
    tso.add_keyword(['ultrasound', 'handheld', '-facial', '-pain', '-massager', '-Liposuction']) # let's define all words we would like to have a look for
    #tso.set_language('de') # we want to see German tweets only
    tso.set_include_entities(True) # and don't give us all those entity information

    # it's about time to create a TwitterSearch object with our secret tokens
    ts = TwitterSearch(
     )

    for tweet in ts.search_tweets_iterable(tso):
        twText = " "
	twText = remo_na(tweet['text'])

	twID = tweet['id']
	#print '%s' % tweet['created_at']
	tWho = '@' + tweet['user']['screen_name']
	created = datetime.strptime(tweet['created_at'], '%a %b %d %H:%M:%S +0000 %Y')

	query = ("SELECT tID FROM tweets "
	"WHERE tUID LIKE %s ")
	numrows = cursor.execute(query, twID)
	if not(numrows): # (tWho, tWhat, tDate, tUID) 
		data_tweet = (tWho.decode('unicode_escape').encode('ascii','ignore'), twText.encode('ascii', 'ignore'), created, twID )
		cursor.execute(add_tweet, data_tweet)
	if numrows:
		print "Tweet deja enregistre"




except TwitterSearchException as e: # take care of all those ugly errors if there are some
    print(e)

		





db.commit()
cursor.close()
db.close()



t-graph.py
#!/usr/bin/python
import sys
from urllib2 import urlopen
import bs4 as BeautifulSoup
import urllib2
# Requis pour MySQL connection
# Requis pour MySQL connection
import MySQLdb
from tecpass import * 
import string
import re


# USAGE 
# python graph.py > ./Bureau/graph.gdf

cursor=db.cursor()


query = ("SELECT tWho, tWhat FROM tweets ")

cursor.execute(query,)
result = cursor.fetchall()	

# let's clean the data 
paireds = []
tHList = []
tUsers = []
tHashes = []
i = 0
for row in result:
	# for bugbs in my mysql dump due to bad retweets
	 
	usered = row[0].lower()
	tmp = string.replace(row[1], 'RT ', 'RT @') 
	tmp = string.replace(tmp, '@@', '@')
	tmp = tmp.lower()
	users = []
	users = re.findall("@\w+", tmp)
	
	tHashers = re.findall("#\w+", tmp)
	users.append(usered)
	
	users = users + tHashers 
	tHashes.append(tHashers)
	#print users
	paireds.append(users)
	#paireds.append(tHashers)

	i = i +1 
	tUsers.append(usered)
	tHList = tHList + tHashers

#print paireds
#print tUsers
tUserUnique = []
[tUserUnique.append(item) for item in tUsers if item not in tUserUnique]
tH = []
[tH.append(item) for item in tHList if item not in tH]
#print tUserUnique

print 'nodedef> name VARCHAR,label VARCHAR'
for user in tUserUnique:
	print "%s, %s" % (user, user)
for user in tH:
	print "%s, %s" % (user, user)

print 'edgedef> node1 VARCHAR,node2 VARCHAR'
for seque in paireds:
	l = len(seque)
	#print seque
	#print l
	if l>1:
		L = 0
		while L < l-1:
			#print L
			#print seque[L]
			#print seque[b]
			LL = L + 1
			
			while LL < l:
				print "%s, %s" % ( seque[L], seque[LL] )
				LL = LL + 1
			L = L+ 1
	


cursor.close()
db.close()


twitter.py
#!/usr/bin/python
import sys
from TwitterSearch import *
# Requis pour MySQL connection
# Requis pour MySQL connection
import MySQLdb
from datetime import datetime


def remo_na(text):
    return ''.join(i for i in text if ord(i)<128)

db=MySQLdb.connect() 
cursor=db.cursor()

add_tweet = ("INSERT INTO tweets "
               "(tWho, tWhat, tDate, tUID) "
               "VALUES (%s, %s, %s, %s)")

# USAGE
# python patent.py US3541840 US3714817 US3864660

try:
    tso = TwitterSearchOrder() # create a TwitterSearchOrder object
    tso.add_keyword(['ultrasound', 'portable', '-facial', '-pain', '-massager', '-Liposuction']) # let's define all words we would like to have a look for
    #tso.set_language('de') # we want to see German tweets only
    tso.set_include_entities(True) # and don't give us all those entity information

    # it's about time to create a TwitterSearch object with our secret tokens
    ts = TwitterSearch(
     )

     # this is where the fun actually starts :)
 

    for tweet in ts.search_tweets_iterable(tso):
        twText = " "
	twText = remo_na(tweet['text'])

	twID = tweet['id']
	#print '%s' % tweet['created_at']
	tWho = '@' + tweet['user']['screen_name']
	created = datetime.strptime(tweet['created_at'], '%a %b %d %H:%M:%S +0000 %Y')

	query = ("SELECT tID FROM tweets "
	"WHERE tUID LIKE %s ")
	numrows = cursor.execute(query, twID)
	if not(numrows): # (tWho, tWhat, tDate, tUID) 
		data_tweet = (tWho.decode('unicode_escape').encode('ascii','ignore'), twText.encode('ascii', 'ignore'), created, twID )
		cursor.execute(add_tweet, data_tweet)
	if numrows:
		print "Tweet deja enregistre"

	


except TwitterSearchException as e: # take care of all those ugly errors if there are some
    print(e)

		





db.commit()
cursor.close()
db.close()



t-witter.py
#!/usr/bin/python
import sys
from TwitterSearch import *
# Requis pour MySQL connection
# Requis pour MySQL connection
import MySQLdb
from datetime import datetime
from tecpass import * 


def remo_na(text):
    return ''.join(i for i in text if ord(i)<128)


cursor=db.cursor()

add_tweet = ("INSERT INTO tweets "
               "(tWho, tWhat, tDate, tUID) "
               "VALUES (%s, %s, %s, %s)")

# USAGE
# python patent.py US3541840 US3714817 US3864660

try:
    tso = TwitterSearchOrder() # create a TwitterSearchOrder object
    tso.add_keyword(['ultrasound', 'portable', '-facial', '-pain', '-massager', '-Liposuction']) # let's define all words we would like to have a look for
    #tso.set_language('de') # we want to see German tweets only
    tso.set_include_entities(True) # and don't give us all those entity information

    # it's about time to create a TwitterSearch object with our secret tokens


     # this is where the fun actually starts :)
 

    for tweet in ts.search_tweets_iterable(tso):
        twText = " "
	twText = remo_na(tweet['text'])

	twID = tweet['id']
	#print '%s' % tweet['created_at']
	tWho = '@' + tweet['user']['screen_name']
	created = datetime.strptime(tweet['created_at'], '%a %b %d %H:%M:%S +0000 %Y')

	query = ("SELECT tID FROM tweets "
	"WHERE tUID LIKE %s ")
	numrows = cursor.execute(query, twID)
	if not(numrows): # (tWho, tWhat, tDate, tUID) 
		data_tweet = (tWho.decode('unicode_escape').encode('ascii','ignore'), twText.encode('ascii', 'ignore'), created, twID )
		cursor.execute(add_tweet, data_tweet)
	if numrows:
		print "Tweet deja enregistre"

	


except TwitterSearchException as e: # take care of all those ugly errors if there are some
    print(e)

		





db.commit()
cursor.close()
db.close()



t-self.py
#!/usr/bin/python
import sys
from TwitterSearch import *
# Requis pour MySQL connection
# Requis pour MySQL connection
import MySQLdb
from datetime import datetime
from tecpass import * 


def remo_na(text):
    return ''.join(i for i in text if ord(i)<128)


cursor=db.cursor()

add_tweet = ("INSERT INTO tweets "
               "(tWho, tWhat, tDate, tUID) "
               "VALUES (%s, %s, %s, %s)")

# USAGE
# python patent.py US3541840 US3714817 US3864660

try:
    tso = TwitterSearchOrder() # create a TwitterSearchOrder object
    tso.add_keyword(['@echopenorg']) # let's define all words we would like to have a look for
    #tso.set_language('de') # we want to see German tweets only
    tso.set_include_entities(True) # and don't give us all those entity information

    # it's about time to create a TwitterSearch object with our secret tokens


    for tweet in ts.search_tweets_iterable(tso):
        twText = " "
	twText = remo_na(tweet['text'])

	twID = tweet['id']
	#print '%s' % tweet['created_at']
	tWho = '@' + tweet['user']['screen_name']
	created = datetime.strptime(tweet['created_at'], '%a %b %d %H:%M:%S +0000 %Y')

	query = ("SELECT tID FROM tweets "
	"WHERE tUID LIKE %s ")
	numrows = cursor.execute(query, twID)
	if not(numrows): # (tWho, tWhat, tDate, tUID) 
		data_tweet = (tWho.decode('unicode_escape').encode('ascii','ignore'), twText.encode('ascii', 'ignore'), created, twID )
		cursor.execute(add_tweet, data_tweet)
	if numrows:
		print "Tweet deja enregistre"




except TwitterSearchException as e: # take care of all those ugly errors if there are some
    print(e)

		





db.commit()
cursor.close()
db.close()

